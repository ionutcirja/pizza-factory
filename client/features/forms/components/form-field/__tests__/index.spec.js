import React from 'react';
import { shallow } from 'enzyme';
import FormField from '..';

jest.mock('../text', () => () => <input type="text" />);

describe('FormField component', () => {
  let propsToRender;
  
  beforeEach(() => {
    propsToRender = {};
  });
  
  describe('render', () => {
    it('should render a TextField component if type prop value is undefined', () => {
      const wrapper = shallow(<FormField {...propsToRender} />);
      expect(wrapper).toMatchSnapshot();
    });
  
    it('should render a TextField component if type prop value is text', () => {
      propsToRender.type = 'text';
      const wrapper = shallow(<FormField {...propsToRender} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
