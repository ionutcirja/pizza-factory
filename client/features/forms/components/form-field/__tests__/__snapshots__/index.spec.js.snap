// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField component render should render a SelectField component if type prop value is select 1`] = `
<FormikConnect(FieldInner)
  component={[Function]}
  options={
    Array [
      "option 1",
      "option 2",
      "option 3",
    ]
  }
/>
`;

exports[`FormField component render should render a TextField component if type prop value is text 1`] = `
<FormikConnect(FieldInner)
  component={[Function]}
/>
`;

exports[`FormField component render should render a TextField component if type prop value is undefined 1`] = `
<FormikConnect(FieldInner)
  component={[Function]}
/>
`;
